File: infra-rest/src/main/java/be/insaneprogramming/cleanarch/rest/payloadmodel/AddTenantToBuildingJsonPayload.java
package be.insaneprogramming.cleanarch.rest.payloadmodel;

import com.fasterxml.jackson.annotation.JsonCreator;
import com.fasterxml.jackson.annotation.JsonProperty;

public class AddTenantToBuildingJsonPayload {
	private final String name;

	@JsonCreator
	public AddTenantToBuildingJsonPayload(@JsonProperty("name") String name) {
		this.name = name;
	}

	public String getName() {
		return name;
	}
}

File: infra-rest/src/main/kotlin/be/insaneprogramming/cleanarch/rest/payloadmodel/AddTenantToBuildingJsonPayload.kt
package be.insaneprogramming.cleanarch.rest.payloadmodel

import be.insaneprogramming.cleanarch.requestmodel.AddTenantToBuildingRequest
import com.fasterxml.jackson.annotation.JsonProperty

class AddTenantToBuildingJsonPayload(@JsonProperty("name") val name: String) {

    fun toRequest(buildingId: String): AddTenantToBuildingRequest {
        return AddTenantToBuildingRequest(buildingId, name)
    }
}
